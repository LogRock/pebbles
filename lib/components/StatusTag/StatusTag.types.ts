import React from "react";

export interface StatusTagProps {
  label?: string;
  size: "micro" | "small" | "medium" | "large";
  variant: "error" | "warning" | "success" | "neutral";
  icon?: React.ReactNode;
  collapsed?: boolean | false;
}
