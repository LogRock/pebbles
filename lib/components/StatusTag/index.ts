export { default } from "./StatusTag";
export { STATUS_TAG_SIZES, STATUS_TAG_VARIANTS } from "./StatusTag.types";
export type { StatusTagProps } from "./StatusTag.types";
