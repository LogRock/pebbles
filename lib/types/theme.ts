import * as CSS from "csstype";
import {
  STATUS_TAG_SIZES,
  STATUS_TAG_VARIANTS,
} from "../components/StatusTag/StatusTag.types";

import {
  blurTokens,
  colorTokens,
  gradientTokens,
  shadeTokens,
  shadowTokens,
  spacingTokens,
  titleTokens,
  textTokens,
} from "./tokens";

export type Spacings = Record<spacingTokens, CSS.Property.Width>;

type Color = Record<shadeTokens, CSS.Property.Color>;

type Colors = Record<colorTokens, Color>;

export interface TypographyProperties {
  fontFamily: CSS.Property.FontFamily;
  textTransform: CSS.Property.TextTransform;
  desktop: {
    weights: {
      normal: CSS.Property.FontWeight;
      bold: CSS.Property.FontWeight;
      bolder: CSS.Property.FontWeight;
      boldest: CSS.Property.FontWeight;
    };
    fontSize: CSS.Property.FontSize;
    lineHeight: CSS.Property.LineHeight;
    tracking: CSS.Property.LetterSpacing;
  };
  mobile: {
    weights: {
      normal: CSS.Property.FontWeight;
      bold: CSS.Property.FontWeight;
      bolder: CSS.Property.FontWeight;
      boldest: CSS.Property.FontWeight;
    };
    fontSize: CSS.Property.FontSize;
    lineHeight: CSS.Property.LineHeight;
    tracking: CSS.Property.LetterSpacing;
  };
}

type Typography = Record<titleTokens | textTokens, TypographyProperties>;

export type Shadows = Record<shadowTokens, CSS.Property.BoxShadow>;

type Blurs = Record<blurTokens, CSS.Property.Width>;

interface LinearGradient {
  background: CSS.Property.Background;
  colors: CSS.Property.Color[];
  angle: number;
}

type Gradients = Record<gradientTokens, LinearGradient>;

interface ButtonProperties {
  background: CSS.Property.BackgroundColor;
  borderColor: CSS.Property.BorderColor;
  borderWidth: CSS.Property.BorderWidth;
  contentColor: CSS.Property.Color;
  textDecorationLine?: CSS.Property.TextDecorationLine;
  borderRadius?: CSS.Property.BorderRadius;
  boxShadow?: CSS.Property.BoxShadow;
  filter?: CSS.Property.Filter;
}

export interface ButtonVariant {
  primary: ButtonProperties;
  secondary: ButtonProperties;
  tertiary: ButtonProperties;
  outlined: ButtonProperties;
}

interface ButtonSize {
  verticalPadding: CSS.Property.PaddingTop;
  horizontalPadding: CSS.Property.PaddingLeft;
  fontSize: CSS.Property.FontSize;
  iconPadding: CSS.Property.Padding;
  borderRadius: CSS.Property.BorderRadius;
}

interface IconButtonSize {
  paddingTop: CSS.Property.PaddingTop;
  paddingLeft: CSS.Property.PaddingLeft;
  paddingBottom: CSS.Property.PaddingBottom;
  paddingRight: CSS.Property.PaddingRight;
  fontSize: CSS.Property.FontSize;
  borderRadius: CSS.Property.BorderRadius;
}

export interface Buttons {
  fontFamily: CSS.Property.FontFamily;
  fontWeight: CSS.Property.FontWeight;
  outline: CSS.Property.Outline;
  regular: {
    large: ButtonSize;
    medium: ButtonSize;
    small: ButtonSize;
    xSmall: ButtonSize;
    primary: ButtonVariant;
    destructive: ButtonVariant;
    disabled: ButtonVariant;
    rounded: ButtonVariant;
  };
  icon: {
    large: IconButtonSize;
    medium: IconButtonSize;
    small: IconButtonSize;
    xSmall: IconButtonSize;
    primary: ButtonVariant;
    destructive: ButtonVariant;
    disabled: ButtonVariant;
  };
}

export interface Focused {
  boxSizing: CSS.Property.BoxSizing;
  boxShadow: CSS.Property.BoxShadow;
  borderRadius: CSS.Property.BorderRadius;
  border: CSS.Property.Border;
}

export interface InputBox {
  spacing?: CSS.Property.MarginTop;
  padding: CSS.Property.Padding;
  boxSizing: CSS.Property.BoxSizing;
  boxShadow: CSS.Property.BoxShadow;
  borderRadius: CSS.Property.BorderRadius;
  border: CSS.Property.Border;
  margin: CSS.Property.Margin;
  background: CSS.Property.Background;
  height: CSS.Property.Height;
  width: CSS.Property.Width;

  placeholder: {
    color: CSS.Property.Color;
  };

  error: {
    color: CSS.Property.Color;
    border: CSS.Property.Border;
    boxShadow: CSS.Property.BoxShadow;
    hintColor: CSS.Property.Color;
    focused: Focused;
  };
  info: {
    color: CSS.Property.Color;
    border: CSS.Property.Border;
    boxShadow: CSS.Property.BoxShadow;
    hintColor: CSS.Property.Color;
    focused: Focused;
  };
  disabled: {
    boxSizing: CSS.Property.BoxSizing;
    boxShadow: CSS.Property.BoxShadow;
    border: CSS.Property.Border;
  };
  label: {
    color: CSS.Property.Color;
    fontStyle: CSS.Property.FontStyle;
    fontSize: CSS.Property.FontSize;
    fontWeight: CSS.Property.FontWeight;
    lineHeight: CSS.Property.LineHeight;
  };
  helper: {
    color: CSS.Property.Color;
    fontStyle: CSS.Property.FontStyle;
    fontSize: CSS.Property.FontSize;
    fontWeight: CSS.Property.FontWeight;
    lineHeight: CSS.Property.LineHeight;
    margin: CSS.Property.Margin;
  };
  hint: {
    background: CSS.Property.Background;
    top: CSS.Property.Inset;
    right: CSS.Property.Inset;
    margin: CSS.Property.Margin;
  };
}

export interface Select {
  items: {
    hover: {
      background: CSS.Property.Background;
    };
    shadow: "1px 1px 2px rgba(0, 0, 0, 0.3)";
  };
}

export interface CheckBoc {
  background: CSS.Property.Background;
  borderColor: CSS.Property.BorderColor;
  borderWidth: CSS.Property.BorderWidth;
  borderRadius: CSS.Property.BorderRadius;
  borderStyle: CSS.Property.BorderStyle;
}
export interface CheckBox {
  checkMark: {
    color: CSS.Property.Color;
    width: CSS.Property.Width;
    height: CSS.Property.Height;
    top: CSS.Property.Top;
    left: CSS.Property.Left;
  };
  width: CSS.Property.Width;
  height: CSS.Property.Height;
  marginRight: CSS.Property.MarginRight;
  default: CheckBoc;
  checked: CheckBoc;
}

export interface Loader {
  fontFamily: CSS.Property.FontFamily;
  size: CSS.Property.Width;
  background: CSS.Property.Color;
  loaderColor: {
    background: CSS.Property.Color;
    loaderBar: CSS.Property.Color;
  };
}

interface AlertColor {
  descriptionColor: CSS.Property.Color;
  background: CSS.Property.Background;
  borderColor: CSS.Property.Color;
  iconColor: CSS.Property.Color;
  titleColor: CSS.Property.Color;
}

export interface Alert {
  neutral: AlertColor;
  primary: AlertColor;
  warning: AlertColor;
  success: AlertColor;
  destructive: AlertColor;
  title: {
    fontSize: CSS.Property.FontSize;
    fontWeight: CSS.Property.FontWeight;
    lineHeight: CSS.Property.LineHeight;
    height: CSS.Property.Height;
  };
  description: {
    fontSize: CSS.Property.FontSize;
    fontWeight: CSS.Property.FontWeight;
    lineHeight: CSS.Property.LineHeight;
    height: CSS.Property.Height;
  };
  iconSize: CSS.Property.FontSize;
  fontFamily: CSS.Property.FontFamily;
  borderRadius: CSS.Property.BorderRadius;
}

export interface ProgressBar {
  bgColor: CSS.Property.Color;
  sizes: {
    small: CSS.Property.Width;
    medium: CSS.Property.Width;
    large: CSS.Property.Width;
    xLarge: CSS.Property.Width;
  };
  borderRadius: {
    small: CSS.Property.Width;
    medium: CSS.Property.Width;
    large: CSS.Property.Width;
    xLarge: CSS.Property.Width;
  };
  barColor: {
    primary: CSS.Property.Color;
    secondary: CSS.Property.Color;
    error: CSS.Property.Color;
    warning: CSS.Property.Color;
    success: CSS.Property.Color;
    neutral: CSS.Property.Color;
    destructive: CSS.Property.Color;
    destructiveAlt: CSS.Property.Color;
  };
}

export interface ProgressRing {
  ringForegroundColor: {
    primary: CSS.Property.Color;
    secondary: CSS.Property.Color;
    error: CSS.Property.Color;
    warning: CSS.Property.Color;
    success: CSS.Property.Color;
    neutral: CSS.Property.Color;
  };
  ringBackgroundColor: {
    primary: CSS.Property.Color;
    secondary: CSS.Property.Color;
    error: CSS.Property.Color;
    warning: CSS.Property.Color;
    success: CSS.Property.Color;
    neutral: CSS.Property.Color;
  };
}

export interface List {
  item: {
    paddingY: CSS.Property.PaddingLeft;
    paddingX: CSS.Property.PaddingTop;
    internalPadding: CSS.Property.Padding;
    margin: CSS.Property.Margin;
    shadow: CSS.Property.BoxShadow;
    background: {
      regular: CSS.Property.BackgroundColor;
      hover: CSS.Property.BackgroundColor;
      active: CSS.Property.BackgroundColor;
    };
    borderRadius: CSS.Property.BorderRadius;
    label: {
      color: CSS.Property.Color;
      weight: CSS.Property.FontWeight;
    };
    description: {
      color: CSS.Property.Color;
      weight: CSS.Property.FontWeight;
    };
    large: {
      typography: TypographyProperties;
    };
    medium: {
      typography: TypographyProperties;
    };
    small: {
      typography: TypographyProperties;
    };
  };
}

export interface Toggle {
  indicator: {
    color: CSS.Property.Color;
    width: CSS.Property.Width;
    height: CSS.Property.Height;
    left1: CSS.Property.Left;
    left2: CSS.Property.Left;
    top: CSS.Property.Top;
  };
  width: CSS.Property.Width;
  height: CSS.Property.Height;
  borderRadius: CSS.Property.BorderRadius;
  marginRight: CSS.Property.MarginRight;
  default: CheckBoc;
  checked: CheckBoc;
}

interface ColorByVariant {
  error: CSS.Property.Color;
  errorAlt?: CSS.Property.Color;
  warning: CSS.Property.Color;
  warningAlt?: CSS.Property.Color;
  success: CSS.Property.Color;
  neutral: CSS.Property.Color;
}

export interface StatusNumber {
  iconColor: ColorByVariant;
  labelColor: ColorByVariant;
}

export interface StatusTrend {
  iconColor: {
    positive: CSS.Property.Color;
    negative: CSS.Property.Color;
    neutral: CSS.Property.Color;
  };
  labelColor: {
    positive: CSS.Property.Color;
    negative: CSS.Property.Color;
    neutral: CSS.Property.Color;
  };
  fontSize: CSS.Property.FontSize;
  fontWeight: CSS.Property.FontWeight;
}

export interface StatusTag {
  color: {
    background: Record<STATUS_TAG_VARIANTS, CSS.Property.BackgroundColor>;
    content: Record<STATUS_TAG_VARIANTS, CSS.Property.Color>;
  };
  paddingX: Record<STATUS_TAG_SIZES, CSS.Property.Width>;
  paddingY: Record<STATUS_TAG_SIZES, CSS.Property.Width>;
  typography: Record<STATUS_TAG_SIZES, TypographyProperties>;
  radius: Record<STATUS_TAG_SIZES, CSS.Property.BorderRadius>;
}

export interface StatusPercentage {
  fontFamily: CSS.Property.FontFamily;
  fontWeight: CSS.Property.FontWeight;
  fontSize: {
    small: CSS.Property.FontSize;
    medium: CSS.Property.FontSize;
    large: CSS.Property.FontSize;
  };
}

export interface Card {
  fontFamily: CSS.Property.FontFamily;
  fontWeight: CSS.Property.FontWeight;
  fontSize: CSS.Property.FontSize;
}

export interface TabView {
  fontFamily: CSS.Property.FontFamily;
  fontWeight: CSS.Property.FontWeight;
  fontSize: CSS.Property.FontSize;
}

export interface Icons {
  size: CSS.Property.FontSize;
  color: CSS.Property.Color;
  spacing: CSS.Property.Margin;

  menu: {
    active: {
      color: CSS.Property.Color;
    };
  };

  input: {
    helper: {
      color: CSS.Property.Color;
    };
    dropdown: {
      color: CSS.Property.Color;
    };
  };

  button: {
    xSmall: {
      paddingTop: CSS.Property.MarginTop;
      paddingRight: CSS.Property.MarginRight;
      paddingBottom: CSS.Property.MarginBottom;
      paddingLeft: CSS.Property.MarginLeft;
    };
    small: {
      paddingTop: CSS.Property.MarginTop;
      paddingRight: CSS.Property.MarginRight;
      paddingBottom: CSS.Property.MarginBottom;
      paddingLeft: CSS.Property.MarginLeft;
    };
    medium: {
      paddingTop: CSS.Property.MarginTop;
      paddingRight: CSS.Property.MarginRight;
      paddingBottom: CSS.Property.MarginBottom;
      paddingLeft: CSS.Property.MarginLeft;
    };
    large: {
      paddingTop: CSS.Property.MarginTop;
      paddingRight: CSS.Property.MarginRight;
      paddingBottom: CSS.Property.MarginBottom;
      paddingLeft: CSS.Property.MarginLeft;
    };
  };

  table: {
    pageSizeSelector: {
      size: CSS.Property.FontSize;
      spacing: CSS.Property.Margin;
    };
    ordering: {
      spacing: CSS.Property.Margin;
    };
    pagination: {
      spacing: CSS.Property.Margin;
    };
  };

  tabs: {
    spacing: CSS.Property.Margin;
    color: CSS.Property.Color;
    active: {
      color: CSS.Property.Color;
    };
  };

  panel: {
    color: CSS.Property.Color;
    spacing: CSS.Property.Margin;
  };

  alerts: {
    spacing: CSS.Property.Margin;
    destructive: {
      color: CSS.Property.Color;
    };
    warning: {
      color: CSS.Property.Color;
    };
    neutral: {
      color: CSS.Property.Color;
    };
    primary: {
      color: CSS.Property.Color;
    };
    success: {
      color: CSS.Property.Color;
    };
  };

  tags: {
    spacing: CSS.Property.Margin;
    destructive: {
      color: CSS.Property.Color;
    };
    destructiveAlt: {
      color: CSS.Property.Color;
    };
    warning: {
      color: CSS.Property.Color;
    };
    success: {
      color: CSS.Property.Color;
    };
  };

  modal: {
    size: CSS.Property.FontSize;
  };
}

export default interface Theme {
  breakpoints: {
    mobile: string;
    desktop: string;
  };
  name: string;
  spacings: Spacings;
  colors: Colors;
  typography: Typography;
  shadows: Shadows;
  blurs: Blurs;
  gradients: Gradients;
  buttons: Buttons;
  inputBox: InputBox;
  select: Select;
  checkBox: CheckBox;
  loader: Loader;
  alert: Alert;
  progressBar: ProgressBar;
  progressRing: ProgressRing;
  list: List;
  toggle: Toggle;
  statusNumber: StatusNumber;
  statusTag: StatusTag;
  statusPercentage: StatusPercentage;
  statusTrend: StatusTrend;
  card: Card;
  tabView: TabView;
  icons: Icons;
}
