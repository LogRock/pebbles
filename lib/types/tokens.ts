export type spacingTokens =
  | "zero"
  | "xxxxsm"
  | "xxxsm"
  | "xxsm"
  | "xsm"
  | "sm"
  | "md"
  | "big"
  | "xbig"
  | "xxbig"
  | "xxxbig"
  | "lg"
  | "xlg"
  | "xxlg"
  | "xxxlg"
  | "huge"
  | "xhuge"
  | "xxhuge"
  | "xxxhuge";

export type shadeTokens =
  | "50"
  | "100"
  | "200"
  | "300"
  | "400"
  | "500"
  | "600"
  | "700"
  | "800"
  | "900";

export type colorTokens =
  | "neutral"
  | "primary"
  | "secondary"
  | "success"
  | "warning"
  | "destructive"
  | "white"
  | "black";

export type sizeTokens = "xSmall" | "small" | "medium" | "large";

export type weightTokens = "normal" | "bold" | "bolder" | "boldest";

export type titleTokens =
  | "displayLarge"
  | "displaySmall"
  | "headingH1"
  | "headingH2"
  | "headingH3"
  | "headingH4"
  | "headingH5"
  | "headingH6";

export type textTokens =
  | "paragraphLarge"
  | "paragraphMedium"
  | "paragraphSmall"
  | "paragraphXSmall"
  | "overlineLarge"
  | "overlineMedium"
  | "overlineXSmall";

export type shadowTokens =
  | "none"
  | "xSmall"
  | "small"
  | "medium"
  | "large"
  | "xLarge"
  | "xxLarge"
  | "upXSmall"
  | "upSmall"
  | "upMedium"
  | "upLarge"
  | "upXLarge"
  | "upXxLarge";

export type blurTokens = "none" | "small" | "medium" | "large" | "xLarge";

export type gradientTokens =
  | "sunburst"
  | "darkWineSea"
  | "cherryCola"
  | "coldSky"
  | "oliveGarden"
  | "morningSakura"
  | "blueMoon"
  | "synthwave"
  | "teaLeaves"
  | "blueberryWine";

export type widthTokens =
  | "1/12"
  | "2/12"
  | "3/12"
  | "4/12"
  | "5/12"
  | "6/12"
  | "7/12"
  | "8/12"
  | "9/12"
  | "10/12"
  | "11/12"
  | "12/12";
