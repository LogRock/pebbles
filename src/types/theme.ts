import { Property } from "csstype";

interface Typography {
  regular?: {
    color?: Property.Color;
    fontSize?: Property.FontSize;
    fontWeight?: Property.FontWeight;
    textDecoration?: Property.TextDecoration;
  };
  inverted?: {
    color?: Property.Color;
    fontSize?: Property.FontSize;
    fontWeight?: Property.FontWeight;
    textDecoration?: Property.TextDecoration;
  };
}

interface NavItem {
  color?: Property.Color;
  background?: Property.BackgroundColor;
  border?: Property.BorderColor;
}

interface ColorVariations {
  lightest?: Property.Color;
  lighter?: Property.Color;
  light?: Property.Color;
  medium?: Property.Color;
  dark?: Property.Color;
  darker?: Property.Color;
  darkest?: Property.Color;
}

export interface Theme {
  breakpoints: {
    mobile: Property.Width;
    tablet: Property.Width;
    desktop: Property.Width;
    widescreen: Property.Width;
    fullhd: Property.Width;
  };
  colors?: {
    primary?: ColorVariations;
    secondary?: ColorVariations;
    status?: {
      success?: ColorVariations;
      info?: ColorVariations;
      warning?: ColorVariations;
      danger?: ColorVariations;
      neutral?: ColorVariations;
    };
    text?: {
      primary?: ColorVariations;
      secondary?: ColorVariations;
      inverted?: ColorVariations;
    };
  };
  spacings?: {
    xxxs: Property.Width;
    xxs: Property.Width;
    xs: Property.Width;
    sm: Property.Width;
    md: Property.Width;
    lg: Property.Width;
    xl: Property.Width;
    xxl: Property.Width;
    xxxl: Property.Width;
  };
  navbar?: {
    background?: Property.BackgroundColor;
    navigation?: {
      item?: {
        regular?: NavItem;
        active?: NavItem;
        hover?: NavItem;
      };
    };
  };
  typography?: {
    header1?: Typography;
    header2?: Typography;
    header3?: Typography;
    header4?: Typography;
    header5?: Typography;
    header6?: Typography;
    paragraph?: Typography;
    link?: Typography;
  };
}
