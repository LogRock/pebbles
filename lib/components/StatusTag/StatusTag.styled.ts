import styled from "styled-components";

import { OverlineLarge } from "../Typography";
import { StatusTagProps } from "./StatusTag.types";

export const StatusTagWrapper = styled.div<
  Pick<StatusTagProps, "variant" | "size">
>`
  display: flex;
  align-items: center;
  justify-content: center;
  width: max-content;
  height: ${({ theme, size }) => theme.statusTag.tagSize[size]};
  padding: 0 ${({ theme }) => theme.spacings.md};

  border: 1px solid
    ${({ theme, variant }) => theme.statusTag.borderColor[variant]};
  border-radius: 64px;

  background-color: ${({ theme, variant }) =>
    theme.statusTag.backgroundColor[variant]};
`;

export const IconWrapper = styled.div<
  Pick<StatusTagProps, "variant" | "label">
>`
  margin-right: ${({ theme, label }) => (label ? theme.spacings.sm : 0)};
  margin-bottom: -2px;

  color: ${({ theme, variant }) => theme.statusTag.iconColor[variant]};
`;

export const StyledOverlineLarge = styled(OverlineLarge)<
  Pick<StatusTagProps, "variant" | "size">
>`
  color: ${({ theme, variant }) => theme.statusTag.labelColor[variant]};
  font-size: ${({ theme, size }) => theme.statusTag.labelSize[size]};
`;
