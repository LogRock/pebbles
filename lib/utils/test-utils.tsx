/* eslint-disable import/export */
import React from "react";
import { cleanup, render } from "@testing-library/react";
import { afterEach } from "vitest";
import { ThemeProvider } from "styled-components";
import { light as theme } from "../theme";

afterEach(() => {
  cleanup();
});

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
// override render export
export { customRender as render };
