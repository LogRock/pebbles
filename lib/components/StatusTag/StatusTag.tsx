import React, { FC } from "react";

import { StatusTagProps } from "./StatusTag.types";
import {
  IconWrapper,
  StatusTagBackground,
  StyledOverlineLarge,
  StatusTagContainer,
} from "./StatusTag.styled";

const StatusTag: FC<StatusTagProps> = ({
  size = "medium",
  variant = "neutral",
  icon,
  collapsed,
  children,
  ...props
}) => {
  return (
    <StatusTagContainer size={size}>
      <StatusTagBackground
        variant={variant}
        size={size}
        collapsed={collapsed}
        {...props}
      >
        {icon && (
          <IconWrapper variant={variant} hasLabel={Boolean(children)}>
            {icon}
          </IconWrapper>
        )}
        <StyledOverlineLarge
          variant={variant}
          size={size}
          collapsed={collapsed}
        >
          {children}
        </StyledOverlineLarge>
      </StatusTagBackground>
    </StatusTagContainer>
  );
};

export default StatusTag;
