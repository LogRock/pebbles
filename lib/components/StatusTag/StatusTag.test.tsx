import React from "react";

import "@testing-library/jest-dom";
import { render } from "../../utils/test-utils";
import StatusTag from ".";
import { STATUS_TAG_SIZES, STATUS_TAG_VARIANTS } from "./StatusTag.types";

describe("StatusTag", () => {
  it("renders properly", () => {
    const { container } = render(
      <StatusTag
        size={STATUS_TAG_SIZES.MEDIUM}
        variant={STATUS_TAG_VARIANTS.DESTRUCTIVE}
      >
        High Risk
      </StatusTag>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
