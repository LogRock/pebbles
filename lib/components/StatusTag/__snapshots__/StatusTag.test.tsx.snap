// Vitest Snapshot v1

exports[`StatusTag > renders properly 1`] = `
<div
  class="sc-bqiRlB lpmcWM"
>
  <div
    class="sc-ksdxgE laSaIJ"
    label="High Risk"
  >
    <p
      class="sc-iqseJM sc-fotOHu xyHVy eQnoQT"
    />
  </div>
</div>
`;

exports[`StatusTag > should render Notification Status 1`] = `
<div
  class="sc-bqiRlB lowtcP"
>
  <div
    class="sc-ksdxgE gjGcRT"
  >
    <p
      class="sc-iqseJM sc-fotOHu xyHVy ewSGrG"
    />
  </div>
</div>
`;
